.gitignore

```diff
@@ -11,3 +11,10 @@ source/src/cip_objects/
 CMakeCache.txt
 *.patch
 .idea/
+*.a
+*.cmake
+Makefile
+CMakeFiles
+devicedata.h
```

Comment:
I don't understand the question. 

What is meant by in-source builds? 

I made changes to .gitignore based on what git identified as "un-tracked" files, and my perception that these were build generated files that should be ignored.

Reply:
In-source builds refers to the approach, to generate the corresponding makefiles inside the source/src folders. Out-Source build is my standard approach, where the generated files are all located in the bin folder. As the bin folder is excluded, I don't need the gitignore entries you added, as all files in bin are already exlcuded by it

Reply:
I see. I'd have to go through the build process again, but to the best of my recollection I did an In-source build because the Out-source build did not work. So if the In-Source build is done, it generates many files that should also be ignored, and these are what I was trying to ignore.